src/main/java/seedu/budgetbuddy/commons/SavingList.java

```diff
@@ -180,6 +177,32 @@ public void addSaving(String category, String amount) throws BudgetBuddyExceptio
     
     
 
+    public void reduceSavingsByCategory(String category, double amount) {
+        List<Saving> matchedSavings = savings.stream()
+                .filter(s -> s.getCategory().equalsIgnoreCase(category))
+                .collect(Collectors.toList());
+
+        if (matchedSavings.isEmpty()) {
+            System.out.println("No savings found under category: " + category);
+            return;
+        }
+
+        boolean allReductionsSuccessful = true;
+        for (Saving saving : matchedSavings) {
+            if (saving.getAmount() >= amount) {
+                saving.setAmount(saving.getAmount() - amount);
+            } else {
+                System.out.println("Insufficient amount in " + category + " to reduce by $" + amount);
```

Comment:
method is well written and easy to understand! Would it be good to include logger statements with the print statements to record the errors caught for more efficient debugging?